tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -226,3 +282,26 @@ def test_cpu_template(test_microvm_with_ssh, network_config, cpu_template):
 
     utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
                                    expected_cpu_features)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   feature_info_1_edx)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   feature_info_1_ecx)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   thermal_and_power_mgmt)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   extended_features)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   xsave_0xd_0)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   xsave_0xd_1)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   extended_080000001_edx)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   intel_080000001_ecx)
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
+                                   adv_pwr_mgmt)
+    if cpu_template != "C3":
```

Comment:
I am probably missing something but since this condition will be valid for a T2, why  are you updating the `c3_masked_features` vector ?

Reply:
because I did want the big lists of features to be generic, and did not want to create another dictionary with masked features from C3, will create one for better readability

Reply:
done.